tests/testsuite/package.rs

```diff
@@ -3304,3 +3304,57 @@ fn init_and_add_inner_target(p: ProjectBuilder) -> ProjectBuilder {
     .file("derp/target/foo.txt", "")
     .file("derp/not_target/foo.txt", "")
 }
+
+#[cargo_test]
+fn custom_build_warning() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+        [package]
+        name = "foo"
+        version = "0.0.1"
+        license = "MIT"
+        description = "foo"
+        authors = []
+        build = "../t_custom_build/custom_build.rs"
+        "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+    p.cargo("package -l")
+        .with_stderr(format!(
+            "\
+warning: manifest has no documentation, homepage or repository.
+See https://doc.rust-lang.org/cargo/reference/manifest.html#package-metadata for more info.
+warning: build `{}/../t_custom_build/custom_build.rs` does not appear to exist.
+Please update the build setting in the manifest at `{}/Cargo.toml`
+This may become a hard error in the future.
+",
+            p.root().display(),
+            p.root().display()
+        ))
```

Comment:
You can use the handy `[CWD]` placeholder here:

```suggestion
        .with_stderr(
            "\
[WARNING] manifest has no documentation, homepage or repository.
See https://doc.rust-lang.org/cargo/reference/manifest.html#package-metadata for more info.
[WARNING] build `[CWD]/../t_custom_build/custom_build.rs` does not appear to exist.
Please update the build setting in the manifest at `[CWD]/Cargo.toml`
This may become a hard error in the future.
",
        )
```

More placeholders can be found in https://github.com/rust-lang/cargo/blob/43fbbc9b36614360eeddefa2df165ecad9ba0e50/crates/cargo-test-support/src/compare.rs#L189-L241

Reply:
nice suggestion.